core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -32,13 +35,24 @@ sealed trait Entity[+F[_]] {
 }
 
 object Entity {
-  def apply[F[_]](body: EntityBody[F], length: Option[Long] = None): Entity[F] =
+
+  @deprecated("Use Entity.stream instead", "1.0.0-M38")
+  def apply[F[_]](body: Stream[F, Byte], length: Option[Long] = None): Entity[F] =
+    stream(body, length)
+
+  def stream[F[_]](body: Stream[F, Byte], length: Option[Long] = None): Entity[F] =
     Default(body, length)
 
   // The type parameter aids type inference in Message constructors.
   def empty[F[_]]: Entity[F] = Empty
   def strict(bytes: ByteVector): Entity[Pure] = Strict(bytes)
 
+  /** Generates a Strict entity with the string encoded the given charset,
+    * which is UTF_8 by default.
+    */
+  def string(str: String, charset: JavaCharset = StandardCharsets.UTF_8): Entity[Pure] =
```

Comment:
I have somehow misseen that you used Java's `Charset` here. We have a dedicated type `org.http4s.Charset`. Let's use it instead, wdyt?

Reply:
This was just replacing what we had in, e.g., the `Message` file, where it is currently using the Java one. Perhaps that may be another PR?

Reply:
Although I'm not against doing it in separate PR, we shouldn't release this version of `Entity.string` basically. I'm in favor of this refactoring overall.

Reply:
Done